### ISSUE TITLE
Error while updating property 'endPoint' of a view managed by: BVLinearGradient

### ISSUE THREAD

**Issue by kildip** [react-native-linear-gradient/react-native-linear-gradient#612]
Hello,

I am not able to run the project because I am facing an issue with the **Error while updating property 'endPoint' of a view managed by: BVLinearGradient**
version:-^2.5.6
`<LinearGradient
          start={{ x: 0.8, y: 0 }}
          end={{ x: 0.4, y: 0.3 }}
          colors={['rgb(247,79,33)', 'rgb(244,160,47)']}
          style={{
            paddingBottom: 15,
            width: '100%',
            paddingLeft: '5%',
            paddingRight: '5%',
            flexDirection: 'column',
          }}
        >`</LinearGradient>`

**Comment by johnnywang** [comment]
I'm experiencing the same, but with start point instead: `JSApplicationIllegalArgumentException
Error while updating property 'startPoint' of a view managed by: BVLinearGradient`

I don't know _how_ this is broken, but this seems to happen when we push OTA updates to our app while we're on the latest version, but is fine if we version lock back to 2.6.2. I suspect it's related to one of the most two recent patches, possibly `Fix start/end point/position naming inconsistency by @friederbluemle in https://github.com/react-native-linear-gradient/react-native-linear-gradient/pull/604`?

**Comment by friederbluemle** [comment]
Hi @kildip - Thanks for the report!

> I am not able to run the project [...]

By "the project", are you referring to the example project in the app or a different project?

> version:-^2.5.6

`^2.5.6` is a version range. Can you specify the exact version you are using? I assume it is one of new 2.7.* versions?

I copy+pasted your code snippet into the latest example app and it appears to work fine for me:

![image](https://github.com/react-native-linear-gradient/react-native-linear-gradient/assets/743291/757311d8-1899-46ab-89ec-a36892df6845)

**Comment by friederbluemle** [comment]
@johnnywang - Thanks for the additional details. You are correct, the error message indeed looks like it is directly related to those commits. Specifically the parameter type change of the two `@ReactProp` annotated methods.
What kind of OTA updates are you pushing to your app? Whenever there are (local) updates to dependencies _that include native modules_ JavaScript code should not be pushed out-of-band to those applications. That's because of the internal contract in the library between the native and JavaScript side (the public interface of the JS view did not change). The safer option is to do a full app update whenever there are native module changes.
@Sunbreak is that also your understanding? Please correct me if I'm wrong or if there is anything to add.

**Comment by johnnywang** [comment]
Hey @friederbluemle, we're using expo-updates to support our OTA updates, so they're just for client-side only updates. Native still goes through the regular full build, store submission workflow. What I was finding though was that I could do a full native build, change a single line of client code as simple as basic copy, do an OTA publish, and encounter a crash with the aforementioned error the moment the update got picked up

**Comment by jvictor98** [comment]
I'm facing the same problem, I didn't want to stop using the library

**Comment by Sunbreak** [comment]
> Sunbreak

Yep. It is supposed to be a `BREAKING CHANG`

**Comment by andy7076** [comment]
same issue, the auther update 2.7.0 at 4 days ago. lock the old version no. maybe complete the problem.

**Comment by johnnywang** [comment]
> Yep. It is supposed to be a `BREAKING CHANG`

@Sunbreak @friederbluemle, if this was intended to be a breaking change, then it seems like this was versioned incorrectly and should have been a major 3.0.0 version, not a minor/patch, assuming this project is following semver. This ends up unexpectedly breaking anyone with minor and patch version ranges set in their package

And to be clear, this breaks even if I've already pushed out a native build containing this version update, then follow up with an OTA update that contains something as simple as a string copy change

**Comment by friederbluemle** [comment]
No, this was definitely _not_ supposed to be a breaking change! 2fb8afc4df5427880de924f0d4bc06c1f76606c9 modified the parameter type of two native Java methods. Those methods are invoked by the JS part of the library, not by user code, thus this _should_ not be a breaking change (the library interface to the user did not change).

The only possible explanation for this crash is a state where the JS and native part of the library are no longer in sync.

> And to be clear, this breaks even if I've already pushed out a native build containing this version update

I'm very interested to understand how this is possible.. Please share more details if you can.

What I _am_ able to reproduce is the case where JS/Native get out-of-sync and result in the crash as described in this thread. As mentioned above, in any production environment, updates to dependencies that include native modules must always be followed by a full native app update. Any other process is risky, as there might be internal changes that are not fully compatible.

All this said, since there are multiple people reporting this issue, I think it is better to revert the patch on the default branch for now, and reapply it once the other 3.0.0 related PRs are ready and merged. It is definitely a prerequisite for supporting the new architecture, because the code generator uses a _single_ source of truth for methods and types (via a [JavaScript specification](https://reactnative.dev/docs/new-architecture-library-intro#define-specs-in-javascript)).

**Comment by friederbluemle** [comment]
https://github.com/react-native-linear-gradient/react-native-linear-gradient/releases/tag/v2.7.3 is now available with 2fb8afc4df5427880de924f0d4bc06c1f76606c9 reverted.

**Comment by jvictor98** [comment]
Thank you!!!

**Comment by johnnywang** [comment]
> I'm very interested to understand how this is possible.. Please share more details if you can.

I haven't tried reproing this against a simpler new app (and I'm not sure there's an easy way to, unfortunately), but the process in our existing app that I used to repro this was:

1. Pin linear gradient to the latest version (2.7.2)
2. Do a native build using `eas build`
3. Install native build to phone, open up the app and ensure that it's actually running
4. Make a small copy change, like "Hello world" -> "Goodbye world"
5. Push an OTA using `eas update`
6. Observe the app crashing when it tries to apply the downloaded update
